hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/common/HoodieSparkEngineContext.java

```diff
@@ -211,4 +218,26 @@ public List<Integer> removeCachedDataIds(HoodieDataCacheKey cacheKey) {
       return removed == null ? Collections.emptyList() : removed;
     }
   }
+
+  @Override
+  public void cancelJob(String groupId) {
+    javaSparkContext.cancelJobGroup(groupId);
+  }
+
+  @Override
+  public void cancelAllJobs() {
+    javaSparkContext.cancelAllJobs();
+  }
+
+  public SparkConf getConf() {
+    return javaSparkContext.getConf();
+  }
+
+  public Configuration hadoopConfiguration() {
+    return javaSparkContext.hadoopConfiguration();
```

Comment:
We do have a serializableHadoopConf in HoodieEngineContext. we should try to use that instead of fetching from jsc here.